tools/src/main/java/org/apache/kafka/tools/ConnectPluginPath.java

```diff
@@ -0,0 +1,399 @@
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class ConnectPluginPath {
+
+    private static final String MANIFEST_PREFIX = "META-INF/services/";
+
+    public static void main(String[] args) {
+        Exit.exit(mainNoExit(args, System.out, System.err));
+    }
+
+    public static int mainNoExit(String[] args, PrintStream out, PrintStream err) {
+        ArgumentParser parser = parser();
+        try {
+            Namespace namespace = parser.parseArgs(args);
+            Config config = parseConfig(parser, namespace, out, err);
+            runCommand(config);
+            return 0;
+        } catch (ArgumentParserException e) {
+            parser.handleError(e);
+            return 1;
+        } catch (TerseException e) {
+            err.println(e.getMessage());
+            return 2;
+        } catch (Throwable e) {
+            err.println(e.getMessage());
+            err.println(Utils.stackTrace(e));
+            return 3;
+        }
+    }
+
+    private static ArgumentParser parser() {
+        ArgumentParser parser = ArgumentParsers.newArgumentParser("connect-plugin-path")
+            .defaultHelp(true)
+            .description("Manage plugins on the Connect plugin.path");
+
+        ArgumentParser listCommand = parser.addSubparsers()
+            .description("List information about plugins contained within the specified plugin locations")
+            .dest("subcommand")
+            .addParser("list");
+
+        ArgumentParser[] subparsers = new ArgumentParser[] {
+            listCommand,
+        };
+
+        for (ArgumentParser subparser : subparsers) {
+            ArgumentGroup pluginProviders = subparser.addArgumentGroup("plugin providers");
+            pluginProviders.addArgument("--plugin-location")
+                .setDefault(new ArrayList<>())
+                .action(Arguments.append())
+                .help("A single plugin location (jar file or directory)");
+
+            pluginProviders.addArgument("--plugin-path")
+                .setDefault(new ArrayList<>())
+                .action(Arguments.append())
+                .help("A list of locations containing plugins");
+
+            pluginProviders.addArgument("--worker-config")
+                .setDefault(new ArrayList<>())
+                .action(Arguments.append())
+                .help("A Connect worker configuration file");
+        }
+
+        return parser;
+    }
+
+    private static Config parseConfig(ArgumentParser parser, Namespace namespace, PrintStream out, PrintStream err) throws ArgumentParserException, TerseException {
+        Set<Path> locations = parseLocations(parser, namespace);
+        switch (namespace.getString("subcommand")) {
+            case "list":
+                return new Config(Command.LIST, locations, out, err);
+            default:
+                throw new ArgumentParserException("No subcommand specified", parser);
```

Comment:
If I don't specify a subcommand, I get a different error. Unless there's a different scenario that can lead to this code path being reached, I don't think we should assume that the root cause will be a lack of subcommand, since users may be frustrated if they see this error message despite specifying a subcommand.

```suggestion
        String subcommand = namespace.getString("subcommand");
        if (subcommand == null) {
            throw new ArgumentParserException("No subcommand specified", parser);
        }
        switch (subcommand) {
            case "list":
                return new Config(Command.LIST, locations, out, err);
            default:
                throw new ArgumentParserException("Unrecognized subcommand: '" + subcommand + "'", parser);
```